app/pages/project/workflow-selection.jsx

```diff
@@ -197,8 +148,10 @@ class WorkflowSelection extends React.Component {
   }
 
   render() {
-    const { loadingSelectedWorkflow } = this.state;
-    return loadingSelectedWorkflow ? <p>Loading workflow</p> : this.props.children;
+    const { children, project, workflow } = this.props;
+    const linkedWorkflows = project.links.workflows || [];
+    const workflowExistsForProject = workflow && linkedWorkflows.indexOf(workflow.id) > -1;
```

Comment:
Should this be inverted and check `workflow.links.project` against the project ID?